modules/tracktion_engine/plugins/external/tracktion_ExternalPlugin.cpp

```diff
@@ -1707,22 +1726,38 @@ bool ExternalPlugin::setBusLayout (juce::AudioChannelSet set, bool isInput, int
 }
 
 //==============================================================================
-juce::String ExternalPlugin::createPluginInstance (const juce::PluginDescription& description)
+void ExternalPlugin::createPluginInstance (const juce::PluginDescription& description)
 {
     jassert (! pluginInstance); // This should have already been deleted!
 
     auto& dm = engine.getDeviceManager();
-
-    juce::String error;
-    pluginInstance = engine.getPluginManager().createPluginInstance (description, dm.getSampleRate(), dm.getBlockSize(), error);
-
-    if (pluginInstance != nullptr)
+    
+    if (requiresAsyncInstantiation (engine, description))
+    {
+        // While instantiation is pending, make sure the plugin is disabled
+        setEnabled (false);
```

Comment:
I'm not sure setting the plugin as "disabled" is the right approach here, that basically means the same thing as bypassed and there's nothing to stop users un-bypassing plugins during the creation time period. It would also show in the UI that these plugins were "bypassed" when they're really just not ready for playback yet.
I think we might need a new state member "isCreatingInstance" to signify this which can be checked at the start of the process block (if that's what the setEnabled is for here?).